src/main/webapp/app/services/visualizations/visualization-grpc.directive.js

```diff
@@ -205,11 +206,22 @@
         function build(forceQuery) {
             if (forceQuery) {
                 angular.element("#loader-spinner").show();
-                proxyGrpcService.forwardCall(vm.datasource.id, {
-                    queryDTO: vm.data.getQueryParameters(filterParametersService.get(), filterParametersService.getConditionExpression(), $rootScope.activePage.activePageNo),
-                    visualMetadata: vm.data,
-                    validationType: 'REQUIRED_FIELDS'
-                }, $stateParams.id === undefined ? $stateParams.viewId : $stateParams.id );
+
+                if (vm.sharefilter) {
+                    proxyGrpcService.forwardCall(vm.datasource.id, {
```

Comment:
Can you handle it in share screen controller itslef instead of changing code here

Reply:
I think no
I  have used the same directive so we can't handle